pyo3-macros-backend/src/module.rs

```diff
@@ -142,12 +145,12 @@ pub fn add_fn_to_module(
 ) -> syn::Result<TokenStream> {
     let mut arguments = Vec::new();
 
-    for (i, input) in func.sig.inputs.iter().enumerate() {
+    for (i, input) in func.sig.inputs.iter_mut().enumerate() {
         match input {
             syn::FnArg::Receiver(_) => {
                 bail_spanned!(input.span() => "unexpected receiver for #[pyfn]");
             }
-            syn::FnArg::Typed(cap) => {
+            syn::FnArg::Typed(ref mut cap) => {
```

Comment:
Is "ref mut" actually needed here? I had a feeling that with default match bindings `cap` will automatically be the mutable reference we want...

Reply:
You're right, thanks! I guess it's one of the remnants of my fight with the borrow checker ;)